### DISCUSSION TITLE
Current status / where to get started contributing?

### DISCUSSION THREAD

**Discussion by wrgrant** [cxw42/TabFern#243] [General]
Hi,

I came poking around yesterday wondering about the current status of the project, as I have been running the alpha 0.3 version for a few months and it has been great. Not a single issue so far.

I’m a developer and was wondering about potentially contributing, but when I looked into the Projects tracker, the three projects in there have a last updated date of 2018.

From my perspective, is a bit difficult for an outsider to browse through the issues, as there are lots of tags, and seeming ties from certain issues back to other blockers.

You do have a WONDERFUL contributing guide. But from a project management “what to work on next” perspective, I’m lost 😅

Maybe a discussion thread could help add a human element to figuring out what is the low hanging fruit to work on for potential contributors? Apologies if this is documented already somewhere.

**Comment by cxw42** [comment]
Thanks for the positive feedback!  Currently, the closest thing to a worklist are the three tags:

- https://github.com/cxw42/TabFern/labels/good%20first%20issue
- https://github.com/cxw42/TabFern/labels/hacktoberfest
- https://github.com/cxw42/TabFern/labels/help%20wanted

Do you have particular specialties or areas of interest?

The biggest long-term task is to move code out of `main_tl.js` into separate modules, and write Jasmine tests for those modules.  It may seem like thankless work (I've done plenty of it!), but that refactoring makes the code much easier to maintain and improve long-term.  One place to start that would be 106--380 ([here](https://github.com/cxw42/TabFern/blob/219dfc4aeeb1d6e15e686b18e06b9c98d792f2cd/app/win/main_tl.js#L106) et seq.).  Those are utility routines that I think could be carved out fairly easily.

**Edit** If you do find an issue you are interested in working on, please comment there and then I will be able to assign it to you.
